ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateRecordFieldsInSignatureFix.fs

```diff
@@ -0,0 +1,85 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.PsiUtil
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+type UpdateRecordFieldsInSignatureFix(error: DefinitionsInSigAndImplNotCompatibleFieldWasPresentError) =
+    inherit FSharpQuickFixBase()
+
+    let getRecordRepresentation (typeDecl: IFSharpTypeDeclaration) =
+        match typeDecl.TypeRepresentation with
+        | :? IRecordRepresentation as rr -> Some rr
+        | _ -> None
+
+    let implementationRecordRepr =
+        let typeDecl = FSharpTypeDeclarationNavigator.GetByIdentifier(error.TypeName)
+        getRecordRepresentation typeDecl
+
+    override this.ExecutePsiTransaction _ =
+        use writeCookie = WriteLockCookie.Create(error.TypeName.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+
+        match implementationRecordRepr with
+        | None -> ()
+        | Some implementationRecordRepr ->
+            let decl = implementationRecordRepr.TypeDeclaration.DeclaredElement
+            let declarations = if isNull decl then Seq.empty else decl.GetDeclarations()
+
+            let signatureRecordRepr =
+                declarations
+                |> Seq.tryPick (fun d ->
+                    match d with
+                    | :? IFSharpTypeDeclaration as signatureTypeDecl when
+                        signatureTypeDecl.GetSourceFile().IsFSharpSignatureFile ->
+                        getRecordRepresentation signatureTypeDecl
+                    | _ -> None)
+
+            match signatureRecordRepr with
+            | None -> ()
+            | Some signatureRecordRepr ->
+                let signatureFieldCount = signatureRecordRepr.FieldDeclarations.Count
+
+                implementationRecordRepr.FieldDeclarations
+                |> Seq.iter (fun implFieldDecl ->
+                    let index = implementationRecordRepr.FieldDeclarations.IndexOf(implFieldDecl)
+                    if index < signatureFieldCount then
+                        // The signature record definition has a field at the current index
+                        // The name or type might be wrong
+                        let signatureFieldDecl = signatureRecordRepr.FieldDeclarations[index]
+                        if implFieldDecl.SourceName = signatureFieldDecl.SourceName &&
+                           implFieldDecl.DeclaredElement.ContainingType = signatureFieldDecl.DeclaredElement.ContainingType then
```

Comment:
This interestingly enough leads to another error:
![image](https://user-images.githubusercontent.com/2621499/192998027-b42be385-3d12-470a-b91c-dc7eb0b53b35.png)

```
error FS0193: The module contains the field    Field2: int    but its signature specifies    Field2: string    The types differ
```

I assume we want to re-use the same quick fix here?

Reply:
I'd probably prefer it updating a single field type only. Probably there should be an option to update one wrong thing and an option to update all, but it's much easier to reason about it after implementing a prototype and seeing if it's as good as you imagined before implementing it. 🙂

Reply:
That sounds reasonable. Could this be part of this PR or would you prefer to have it in another?

Reply:
Updating a single field type sounds like a subtask of the bigger QF, so it's probably fine to have it in this PR. Plus, it'll be easier to dog food it earlier 🙂

Reply:
Alright, thanks! I can probably easily extract a helper function that takes two `IRecordFieldDeclaration` (`sig` and `impl`) and compares/updates accordingly.
This would be re-usable between both quick fixes.
Where would I store this helper function 😸?

Reply:
Let's create a module next to the quick fixes for now.